libthrift update to 0.14.1 in conda-forge broke CI

### Describe the problem
There was a recently pushed update to `libthrift` on conda-forge which bumped its version to `0.14.1`.
This breaks `pyarrow` as it has (for some unknown reason) linked against exactly `0.14.0` instead of linking against `0.14`, so currently tests fail like this:

```
ImportError while loading conftest '/home/runner/work/modin/modin/modin/conftest.py'.
modin/conftest.py:20: in <module>
    import pyarrow.parquet as pq
/usr/share/miniconda/envs/modin/lib/python3.7/site-packages/pyarrow/parquet.py:33: in <module>
    import pyarrow._parquet as _parquet
E   ImportError: libthrift.so.0.14.0: cannot open shared object file: No such file or directory
Error: Process completed with exit code 4.
```

--- a/environment-dev.yml
+++ b/environment-dev.yml
@@ -5,6 +5,7 @@ dependencies:
   - pandas==1.2.3
   - numpy>=1.16.5,<1.20  # pandas gh-39513
   - pyarrow>=1.0.0
+  - libthrift<0.14.1  # ref: modin gh-2840
   - dask[complete]>=2.12.0,<=2.19.0
   - distributed>=2.12.0,<=2.19.0
   - xarray

--- a/requirements/env_omnisci.yml
+++ b/requirements/env_omnisci.yml
@@ -5,6 +5,7 @@ channels:
 dependencies:
   - pandas==1.2.3
   - pyarrow==2.0.0
+  - libthrift<0.14.1  # ref: modin gh-2840
   - numpy>=1.16.5
   - pip
   - pytest>=6.0.1

